### DISCUSSION TITLE
Is B3 still a thing in 2024?

### DISCUSSION THREAD

**Discussion by danielepantaleone** [BigBrotherBot/big-brother-bot#436] [General]
It's been a long time since I last played online video games. Even more since I had spare time to keep B3 updated (sadly 😢). So, I was wondering if B3 is still widely used among game server managers in 2024 or if there are new techs that makes it outdated or not needed anymore.

**Comment by grb-ani** [comment]
I think it's still used by some but less so now than in the past. A lot of newer games don't let you host your own servers or give access to server log files so that rules b3 out...

So an example of a tool that I think a lot of admins have moved onto is IW4MAdmin because it's still actively developed and offers newer features.

I think b3 requiring python 2.7 also doesn't help a whole lot either. It means people have to either manage multiple python versions, or know how to dockerize their systems. (or convert the codebase to python3 which isn't a particularly small job)

As well as that, not having an open forum to discuss ideas discourages usage, for example, someone may have a request for a certain plugin/feature/parser. Hopefully this discussions section will help with that.

I still find b3 tremendously useful and in the past year I've spent a lot of time writing new plugins and finding new ways to use it.

**Reply by danielepantaleone** [reply]
Thanks for the info. Back in 2018 I started converting B3 to python 3 and I wanted to take this opportunity also to give it a more structured architecture and a more user friendly way to customize its functionalities using external plugins. Sadly with work and family going on I didn't manage to find enough time to continue the development, also because I'm not programming with python anymore since years, so I'm kinda "out of date" (doing mobile stuff lately).

**Reply by wickedvixx** [reply]
Good morning, 

I am still playing in the Call of Duty 1 scene (yes 21yrs later)… we us B3 on all our servers and love it.

**Reply by grb-ani** [reply]
> Thanks for the info. Back in 2018 I started converting B3 to python 3 and I wanted to take this opportunity also to give it a more structured architecture and a more user friendly way to customize its functionalities using external plugins. Sadly with work and family going on I didn't manage to find enough time to continue the development, also because I'm not programming with python anymore since years, so I'm kinda "out of date" (doing mobile stuff lately).

I had noticed the v2.0 branch actually. Well that's completely understandable, rewriting b3 from the ground up sounds like a big job especially if it's mostly done by one person.

I've managed to convert (our) b3 so it's working with python3 but I've only modified the parts we use. It does in fact run without issues and at least it lets me write plugins in py3 and rely on external py3 packages.

Regarding this master branch, I'd actually be quite interested to contribute to a python3 effort but I'm not experienced enough to attempt rewriting b3 from the ground up.
